src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +34,13 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+
+        String[] trimmedAddressParts = trimmedAddress.split(", ");
```

Comment:
`trimmedAddressParts` might not have all four parts. If so, it'll cause a null pointer exception somewhere down the line. You can update `ADDRESS_VALIDATION_REGEX` to fix this issue.